src/app/fdctl/fdctl.h

```diff
@@ -68,6 +68,18 @@ typedef union {
   struct {
     char name[ 13UL ];
   } flame;
+
+  struct {
+    const char * affinity;
```

Comment:
These should be arrays in the struct, and copy the string in there. You don't know if lifetime of those arguments.